### PR TITLE
Support --vars replacements in backend config as well as VCL

### PR DESCRIPTION
**Pull request by jonnangle**
This is useful when handling logentries tokens.

### CODE REVIEW

**Comment by jonnangle** [comment]
@matthew-andrews this OK?

File: lib/loadBackends.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+		var val = process.env[key];
+		if( val ) {
+			var regex = new RegExp('\\\$\\\{'+ key.trim()+'\\\}', 'gm');
+			backendData = backendData.replace(regex, val);
+		} else {
+			throw new Error(`Environment variable ${key} is required to deploy backends`);
+		}
+	}
+
+	return JSON.parse(backendData);
```
Comment (matthew-andrews) [inline lib/loadBackends.js]: ahhh, this means that you can't write the backend data in JavaScript…  https://github.com/Financial-Times/next-cdn/blob/master/src/config/prod.js

maybe as simple as adding a line on line 5:-
```js
if (/\.js$/test(backendsFile) {
  return require(path.join(process.cwd(), backendsFile));
}
```

Comment (jonnangle) [inline lib/loadBackends.js]: ah I see - no probs, I'll just switch to use that pattern instead, thanks!

Comment (matthew-andrews) [inline lib/loadBackends.js]: ah yes that would work too

**Pull request closed without merging**